docs/user_guide/examples.md

```diff
@@ -1,14 +1,248 @@
+Instead of having to edit the requirements in all the class methods that depends on this requirement, the ```reqs``` attribute can be simply modified with:
+```python
+MyTask.reqs.RequiredTask = ModifiedRequiredTask
+```
+Therefore, using the class attribute ```reqs```, when working with the requirements within the class is conveninet.
+
+
+![req plot](cf_plot.svg "image")
+
+### RunOnceTask
+
+The {external+law:py:class}`~law.tasks.RunOnceTask` is a special task that is used to run a task only once, which can be very useful in many situations.
+For example, when a task is required by multiple tasks or multiple branches of a workflow, but the task itself should only be executed once independent of the branches, or a task does not produce any file output but is needed to be marked as done (e.g. a task only produces text output on the CLI).
+
+This is implemented by (additionally) inheriting from {external+law:py:class}`~law.tasks.RunOnceTask`.
+Moreover, to mark the task as complete, the {external=law:py:meth}`~law.tasks.RunOnceTask.mark_complete()` method is to be added at the end of the ```run()``` method after the logic of the task is executed, or the complete```run()``` method is either to be wrapped by the {external=law:py:meth}`~law.tasks.RunOnceTask.complete_on_success` decorator.
+One must be careful with the latter, since the decorator will mark the task as complete even if it fails.
+Therefore, it is recommended to also add a check at the end and raise an error in case the task fails.
+
+```python
+import law
+
+from columnflow.tasks.framework.base import AnalysisTask
+
+class MyPrintingTask(AnalysisTask, law.tasks.RunOnceTask):
+
+    @law.tasks.RunOnceTask.complete_on_success
+    def run(self):
+        # Do something here
+        if fail:
+            raise Exception("this task just failed")
+
+    # Alternative
+    def run(self):
+        # Do something here
+        self.mark_complete()
+```
+
+### Add the task to the analysis
+
+After creating a new task, make sure to add the new task to the law config file in your analysis folder, in order to run the task.
+The new task has to be specified under the ```[outputs]``` section as follows:
+
+```ini
+[outputs]
+
+lfn_sources: wlcg_fs_desy_store
+
+cf.MyNewTask: wlcg
+...
+```
+### Executing the Task
+
+after sourcing the law environment, the task can be executed using the command:
+```bash
+law run cf.MyTask --param1 foo
+```
+The parameters can be given using the syntax ```--param1 foo```, where ```param1``` is the name of the parameter and ```foo``` is the value of the parameter.
+If the parameter's name include an underscore, the underscore is replaced by a hyphen (```--param_1 foo``` -> ```--param-1 foo```).
+If the parameter is a boolean, the parameter can be set to ```True``` by only typing ```--param-boolean```.
```

Comment:
It might be safer to use `param_boolean` in your example above, just to make sure that the user doesn't _need_ to append `boolean` to a parameter in order to make it a boolean (sounds dumb, but better save than sorry)